### PR TITLE
feat(tests): expand and harden test suite; lift coverage to ~81%

### PR DESCRIPTION
**Pull request by CommitMaster102**
- Tokenizer: parameterized coverage for snake/camel/acronyms, NFKC, punctuation, mixed scripts; shingle edge-cases.

- Similarity modes: param + charjaccard/winnow wiring and direct checks.

- Progress/timing: flags + Spinner/AutoProgressBar/StepDisplay; humanized durations.

- Orchestrate: scoring for charjaccard/winnow/minhash; error exits; selection priority/recency.

- git_ops: run/ensure_repo/hash/discover/content/blob id/branches/timestamps/remotes; ensure_remote_with_refspec.

- github_api: parsing, HTTP handling, forks pagination/offset/limit, remote collision suffix.

- utils: normalization, repo dir, cache cleanup, Spinner, formatters, progress bar, StepDisplay, StepTimer.

### CODE REVIEW

**Comment by CommitMaster102** [comment]
giphy

**Review by CommitMaster102 — COMMENTED** [review]
All fine

**Review by CommitMaster102 — COMMENTED** [review]
yeah all good

File: tests/test_orchestrate_scoring_and_errors.py [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+            patch("find_commits_lib.utils.cleanup_repo_cache"),
+            patch("find_commits_lib.git_ops.branches_containing", return_value=[]),
+            patch(
+                "find_commits_lib.git_ops.commit_timestamp",
+                side_effect=lambda _r, c: {"best": 30, "mid": 20, "low": 10}[c],
+            ),
+        ):
+            try:
+                orchestrate(args)
+            except SystemExit:
```
Comment (CommitMaster102) [inline tests/test_orchestrate_scoring_and_errors.py:303]: Yeah, all good

**Review by CommitMaster102 — COMMENTED** [review]
no changes
